tasks/collect-data/collect-data.yaml

```diff
@@ -201,3 +204,45 @@ spec:
             "$(workspaces.data.path)/$(params.subdirectory)/release_plan_admission.json"
 
         exit $RC
+    - name: print-pipeline-ref-info
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      script: |
+        #!/usr/bin/env bash
+        set +x
+
+        pipelineref=$(jq -c '.spec.pipeline.pipelineRef' \
+          "$(workspaces.data.path)/$(params.subdirectory)/release_plan_admission.json")
+        resolver=$(jq -r '.resolver // ""' <<< "${pipelineref}")
+        if [ "${resolver}" == "git" ] ; then
+          url=$(jq -r '.params[] | select(.name=="url") | .value' <<< "${pipelineref}")
+          revision=$(jq -r '.params[] | select(.name=="revision") | .value' <<< "${pipelineref}")
+          pathinrepo=$(jq -r '.params[] | select(.name=="pathInRepo") | .value' <<< "${pipelineref}")
+          orgrepo="${url#*/*/*/}"
+          org=$(echo "${orgrepo}" | cut -f1 -d/)
+          repo=$(echo "${orgrepo}" | cut -f2 -d/ | cut -d. -f1)
+
+          sha=$(curl -s "https://api.github.com/repos/${org}/${repo}/commits/${revision}" | jq -r '.sha // ""')
+
+        fi
+
+        org="${org:-unknown}"
+        repo="${repo:-unknown}"
+        revision="${revision:-unknown}"
+        pathinrepo="${pathinrepo:-unknown}"
+        sha="${sha:-unknown}"
+
+        echo ""
+        echo "Release Pipeline Ref Info:"
+        echo "----------------------"
+      
+        json=$(jq -n -c \
+          --arg org "${org}" \
+          --arg repo "${repo}" \
+          --arg revision "${revision}" \
+          --arg pathinrepo "${pathinrepo}" \
+          --arg sha "${sha}" \
+          '$ARGS.named | tostring')
+
+        echo "${json}" > "$(results.releasePipelineMetadata.path)"
+        # pretty print for log message
+        echo "${json//\\/}" 
```

Comment:
How does this make it prettier? It just removes backslashes AFAICS? Why not just do `jq . "${json}"` instead?

Reply:
check what I now. it prints:

```
[run-task : print-pipeline-ref-info] Release Pipeline Ref Info:
[run-task : print-pipeline-ref-info] --------------------------
[run-task : print-pipeline-ref-info] {
[run-task : print-pipeline-ref-info]   "org": "some-catalog",
[run-task : print-pipeline-ref-info]   "repo": "some-service-catalog",
[run-task : print-pipeline-ref-info]   "revision": "development",
[run-task : print-pipeline-ref-info]   "pathinrepo": "pipelines/rh-advisories/rh-advisories.yaml",
[run-task : print-pipeline-ref-info]   "sha": "12345"
[run-task : print-pipeline-ref-info] }
```

Reply:
OK, so I checked again. It seems really funky what's going on.
First you do something like this:
```
$ jq -n -c --arg org myrg --arg repo myrepo '$ARGS.named | tostring'
"{\"org\":\"myrg\",\"repo\":\"myrepo\"}"
```
You save this to a result. Why do you need all the `"` escaped? Why not save a normal json?
```
$ jq -n -c --arg org myrg --arg repo myrepo '$ARGS.named'
{"org":"myrg","repo":"myrepo"}
```
And then you can just do `jq . <<< "$json"` for the pretty output.

Reply:
when I try what you suggest, I get this error in the test:

```
[check-result : check-result] ++ jq -r .org
[check-result : check-result] parse error: Invalid numeric literal at line 1, column 5
```

from here https://github.com/konflux-ci/release-service-catalog/pull/636/files#diff-e43192c80d965fdb7034535453eb6474d7089cd1f0c4e864fe68a18059a6c290R144

I believe the escaped quotes are needed because of how tekton deals with them in results?

Reply:
```
[check-result : check-result] + echo '{org:some-catalog,repo:some-service-catalog,revision:development,pathinrepo:pipelines/rh-advisories/rh-advisories.yaml,sha:12345}
[check-result : check-result] '
[check-result : check-result] {org:some-catalog,repo:some-service-catalog,revision:development,pathinrepo:pipelines/rh-advisories/rh-advisories.yaml,sha:12345}
```

Reply:
OK, so I tried it for myself and I know what the issue is. You can save a normal json in the result, no escaping is needed. The problem is in the check-result task in the test.

When you do this:
```
echo "$(params.releasePipelineMetadata)"
```
Tekton will substitute the `$()` with the parameter value before starting the script. So with a json, you will have:
```
echo "{"org":"myorg"}"
```
You see the problem? I actually never realized it. But when we use params in the scripts, you should probably always wrap it in `''` and not `""`. Even better, and I think it's recommended by Tekton, is to set env vars from the params. The you can work with the env var as usual.

Reply:
I pushed a new commit with the fix.

Reply:
It seems this is addressed.